### PR TITLE
feat: implement TSIGKey API

### PR DESCRIPTION
**Pull request by bennetgallein**
## Description

Add APIs for the /tsigkeys route:
https://doc.powerdns.com/authoritative/http-api/tsigkey.html#tsigkey-endpoints

More information about TSIG:
https://doc.powerdns.com/authoritative/tsig.html

## Motivation and context

TSIGKeys were not implemented

## How has this been tested?

testing was done using the local docker containers

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply.

Please, please, please, don't send your pull request until all of the boxes are ticked. Once your pull request is created, it will trigger a build on our [continuous integration](http://www.phptherightway.com/#continuous-integration) server to make sure your [tests pass](https://help.github.com/articles/about-required-status-checks/).

- [x] I have read the **[CONTRIBUTING](../.github/CONTRIBUTING.md)** document.
- [x] My pull request addresses exactly one patch/feature.
- [x] My pull request contains a title that can be used as a release note.
- [x] I have created a branch for this patch/feature.
- [x] Each individual commit in the pull request is meaningful.
- [x] I have added tests to cover my changes.
- [x] If my change requires a change to the documentation, I have updated it accordingly.

If you're unsure about any of these, don't hesitate to ask. We're here to help!

### CODE REVIEW

**Comment by bennetgallein** [comment]
I'm really struggeling to understand how the API handles updates, like the `PUT` request is currently just creating new keys all the time, even when I only send the `algorithms` field.. I've pushed this to my branch here: https://github.com/bennetgallein/powerdns-php/tree/feat/tsig-keys-debug in case someone can help me out getting this working the right way..

**Comment by bennetgallein** [comment]
Hey Tristan! @trizz  

I think I've figured it out - there are some quirks with the way the API works that (imo) don't really make sense. I'll list them below to shed some light:

- Whenever you send the `name` field, powerdns will clone the key with `id` and create a copy with the new name, but also giving it a new `id`, since the `id` seems to be based on the `name` (just a dot appended(?)).
- when changing the algorithm the key doesn't change, but when creating it get's generated.. It kinda makes sense but why would one want to update the algo without regenerating the key? :D  - I did not test changing the algo and sending an empty key to see if that would trigger a regenerate tho..

I think that's it - lmk if you have any feedback or questions :)

File: src/Resources/TSIGKey.php [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+     * @param string $type Set to "TSIGKey"
+     *
+     * @return self
+     */
+    public function setType(string $type)
+    {
+        $this->type = $type;
+
+        return $this;
+    }
```
Comment (trizz) [inline src/Resources/TSIGKey.php:82]: Should it be possible to change the type?

**Comment by trizz** [comment]
> there are some quirks with the way the API works that (imo) don't really make sense

Yup, some things don't make any sense.

> Whenever you send the name field, powerdns will clone the key with id and create a copy with the new name, but also giving it a new id, since the id seems to be based on the name 

Yes, updating records happens in a similar fashion. When updating a record, PowerDNS deletes and re-adds the whole record. However, because no "real" IDs are returned by the PowerDNS API, this isn't very noticeable. I've tried some of your created methods and they seem to work fine. We don't use TSIG keys ourselves (hence it is missing), so I cannot test with real-world data.

> just a dot appended

Probably to get a canonical domain. The same is required for zones, for example.

So, everything looks fine to me. If no other corrections are necessary (see comment on your code), I'll merge and release this feature.

File: src/Resources/TSIGKey.php (hunk 1 above)
Comment (bennetgallein) [reply src/Resources/TSIGKey.php:82]: the type is changeable, but seems pretty fixed in the documentation i'll remove the setter :)

**Pull request merged**